.github/actions/deploy-with-medic-conf/action.yml

```diff
@@ -0,0 +1,33 @@
+name: 'Deploy with medic-conf'
+inputs:
+  directory:
+    description: The path to the directory containing the configuration to be deployed.
+    required: true
+    default: '.'
+  
+  hostname:
+    description: The hostname of the instance to receive the configuration.
+    required: true
+
+  username:
+    description: The username of an administrator account on the instance at "inputs.hostname". This user will make the deployment.
+    required: true
+    default: 'medic'
+
+  password:
+    description: The password for user with name "inputs.username"
+    required: true
+runs:
+  using: 'composite'
+  steps:
+  - name: install dependencies
+    run: |
+        npm ci
+        pip install --user -U pip
+        sudo python -m pip install git+https://github.com/medic/pyxform.git@medic-conf-1.17#egg=pyxform-medic
+    shell: bash
+    working-directory: ${{ inputs.directory }}
+  - name: run medic-conf 
+    run: npx medic-conf --url=https://${{ inputs.username }}:${{ inputs.password }}@${{ inputs.hostname }} compile-app-settings convert-app-forms convert-collect-forms convert-contact-forms upload-app-settings upload-app-forms upload-collect-forms upload-contact-forms upload-resources upload-custom-translations --force
```

Comment:
Is this list of commands universal? Shouldn't we just make a new command in medic-conf as an alias for all of this stuff? This would allow new commands to be added without having to modify this script and potentially break backwards compatibility.

Reply:
i would welcome suggestions for further improvements, but don't think that should block this PR.

Reply:
Yeah fair enough, no need to block this PR on a medic-conf release. I've added [a comment](https://github.com/medic/medic-conf/issues/312#issuecomment-736812994) to the relevant medic-conf issue to capture this, we'll just have to remember to come back and update this script later...